run.py

```diff
@@ -867,9 +872,12 @@ def main():
             cwd, dirpath, outdir, suricata_config, args.verbose)
         try:
             results = test_runner.run()
-            passed += results["success"]
-            failed += results["failure"]
-            skipped += results["skipped"]
+            if results["failure"] > 0:
+                failed += 1
+            elif results["skipped"] > 0:
+                skipped += 1
+            elif results["success"] > 0:
+                passed += 1
```

Comment:
I mean that we should handle the default case somehow, even if we are never supposed to get there

Reply:
Could throw an exception here. Its not a state we ever get in right now, doing so would likely be an error in an update to run.py.

Reply:
So are we counting the number of checks in the test.yaml that actually passed? But not doing anything with that info? I wonder if we can simplify to just success/fail/skip as booleans rather than counters.

Reply:
> Could throw an exception here

Good idea

> But not doing anything with that info? I wonder if we can simplify to just success/fail/skip as booleans rather than counters.

I would like to do something with that info (even if we do not do it yet)

Reply:
> > Could throw an exception here
> 
> Good idea
> 
> > But not doing anything with that info? I wonder if we can simplify to just success/fail/skip as booleans rather than counters.
> 
> I would like to do something with that info (even if we do not do it yet)

Should I keep that info then or discarding seems like the better thing to do?

Reply:
We can probably keep it.  Maybe add it to the individual test output at some point. For example:
```
===> test-bad-quotation-marks-rule-1: OK (checks: 3, skipped: 1)
```

Reply:
> Could throw an exception here. Its not a state we ever get in right now, doing so would likely be an error in an update to run.py.

Right. In fact, I just ran into a situation like that and the traceback seemed to be a lot more helpful than a generic exception message. Shouldn't it actually fail if there are problems with the script instead of catching the exceptions? Please let me know.

Reply:
> > Could throw an exception here. Its not a state we ever get in right now, doing so would likely be an error in an update to run.py.
> 
> Right. In fact, I just ran into a situation like that and the traceback seemed to be a lot more helpful than a generic exception message. Shouldn't it actually fail if there are problems with the script instead of catching the exceptions? Please let me know.

I think we should have an exception here.  We could get here with valid code, but a programming logic error.  So we should throw a type of exception that will result in a traceback.

But getting this PR in as-is is more important than this IMO.

Reply:
OK thanks. Leaving this as-is then.